tests/nested/manual/core20-cloud-init-maas-signed-seed-data/task.yaml

```diff
@@ -139,53 +167,119 @@ execute: |
   . "$TESTSLIB/nested.sh"
 
   echo "Waiting for snapd to react to cloud-init"
-  seen=0
-  # the first message is about snapd noticing that cloud-init is in error, and
-  # for how long snapd will wait for cloud-init, this message should show up
-  # relatively quickly
-  for i in $(seq 60); do
-      if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "cloud-init reported to be in error state, will disable in 3 minutes"; then
-          seen=1
-          break
-      fi
-      sleep 1
-  done
-  if [ "$seen" = "0" ]; then
-      echo "snapd did not notice cloud-init in error state, test failed"
-      exit 1
-  fi
-  seen=0
-
-  # we now wait for 3 minutes before snapd gives up waiting and disables 
-  # cloud-init
-  for i in $(seq 200); do
-      if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "reported to be in error state after 3 minutes"; then
-          seen=1
-          break
-      fi
-      sleep 1
-  done
-
-  if [ "$seen" = "0" ]; then
-      echo "snapd did not disable cloud-init, test failed"
-      exit 1
+
+  # due to a cascade of failures, nothing works correctly when nesting/combining
+  # tests.nested with retry, so just give up entirely and manually copy all the
+  # commands we want to run below to individual scripts and execute them by 
+  # themselves, I literally probably cannot touch this code again without crying
+  cat > /tmp/snapd-journal-match-script-disable-state.sh << 'EOF'
+  #!/bin/bash -e
+  sudo journalctl --no-pager -u snapd | MATCH "cloud-init reported to be in disabled state"
+  EOF
+
+  cat > /tmp/snapd-journal-match-script-error-notice-state.sh << 'EOF'
+  #!/bin/bash -e
+  sudo journalctl --no-pager -u snapd | MATCH "cloud-init reported to be in error state, will disable in 3 minutes"
+  EOF
+
+  cat > /tmp/snapd-journal-match-script-error-disable-state.sh << 'EOF'
+  #!/bin/bash -e
+  sudo journalctl --no-pager -u snapd | MATCH "cloud-init reported to be in error state after 3 minutes"
+  EOF
+
+  tests.nested copy /tmp/snapd-journal-match-script-disable-state.sh
+  tests.nested copy /tmp/snapd-journal-match-script-error-notice-state.sh
+  tests.nested copy /tmp/snapd-journal-match-script-error-disable-state.sh
+  tests.nested exec "chmod +x /home/user1/snapd-journal-match-script-disable-state.sh"
```

Comment:
Shouldn't we have a bash function that does all this (create the script, copyit, make it executable and run it)?

Reply:
Maybe that would make sense, but I am not going to add that to this PR

Reply:
Yes, that's fine.